src/Auth0.AuthenticationApi/AuthenticationApiClient.cs

```diff
@@ -285,6 +285,32 @@ public Task<AccessTokenResponse> GetTokenAsync(PasswordlessSmsTokenRequest reque
                 body);
         }
 
+        /// <inheritdoc/>
+        public async Task<AccessTokenResponse> GetTokenAsync(DeviceCodeTokenRequest request)
+        {
+            if (request == null) throw new ArgumentNullException(nameof(request));
+
+            var body = new Dictionary<string, string>
+            {
+                {"grant_type", "urn:ietf:params:oauth:grant-type:device_code"},
+                {"device_code", request.DeviceCode},
+                {"client_id", request.ClientId}
+            };
+
+            // TODO: This will throw an exception while the user is completing their part of the flow. From what
+            // I remember the "error" is recoverable via ErrorApiException.ApiError.Error but not "error_description"
+            var response = await connection.SendAsync<AccessTokenResponse>(
+                HttpMethod.Post,
+                tokenUri,
+                body
+            ).ConfigureAwait(false);
```

Comment:
Would you think the SDK should handle this Error? I am inclined to keep it out of the SDK.

Reply:
Not so much handling, but the API returns `error_description` which wouldn't be parsed by `ApiError`.

`ApiError` would need something like `[JsonProperty("error_description")] public string ErrorDescription { get; set; }` to be able to capture it.

I haven't really got a deep enough understanding of the vision behind the SDK to know if this is a good idea or not.

Reply:
This looks like a general problem with the Errors getting returned by the API, not just in this case. So for this PR I would not worry about that.

Reply:
@acraven The error_description is captured as part of the Message property: https://github.com/auth0/auth0.net/blob/master/src/Auth0.Core/Serialization/ApiErrorConverter.cs#L17